src/software/jetson_nano/thunderloop.cpp

```diff
@@ -135,14 +136,14 @@ Thunderloop::~Thunderloop() {}
 
                 // Update primitive executor's primitive set
                 {
-                    struct timespec time_since_last_primitive_received;
                     clock_gettime(CLOCK_MONOTONIC, &current_time);
                     ScopedTimespecTimer::timespecDiff(
                         &current_time, &last_primitive_received_time,
                         &time_since_last_primitive_received);
                     clock_gettime(CLOCK_MONOTONIC, &last_primitive_received_time);
                     network_status_.set_ms_since_last_primitive_received(
-                        time_since_last_primitive_received.tv_nsec / 1000000);
+                        (uint32_t)(time_since_last_primitive_received.tv_nsec /
+                                   NANOSECONDS_PER_MILLISECOND));
```

Comment:
We should use C++ style casting instead of C style. (i.e. using `static_cast<uint32_t>`). Same with other casts